src/test/java/datadriven/TestAppium.java

```diff
@@ -0,0 +1,99 @@
+package datadriven;
+
+import io.appium.java_client.MobileElement;
+import io.appium.java_client.android.AndroidDriver;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class TestAppium {
+    private AndroidDriver driver;  //global variable
+
+    //external data source
+    ReadData excel = new ReadData("TestDataFile/datadriven.xlsx");
+
+    @Before
+    public void setUp() throws MalformedURLException {
+
+        System.out.println("now the shows begins");
+
+        //appium desired capabilities start here
+        //essential desired capabilities for setup the connection between appium server and emulator device
+        DesiredCapabilities desiredCapabilities = new DesiredCapabilities();
+
+        desiredCapabilities.setCapability("platformName", "Android");
+
+        desiredCapabilities.setCapability("appium:platformVersion", "11.0");
+
+        desiredCapabilities.setCapability("appium:deviceName", "4.7_WXGA_API_30");
+
+        desiredCapabilities.setCapability("appium:automationName", "appium");
+
+        desiredCapabilities.setCapability("appium:noRest", true);
+
+        desiredCapabilities.setCapability("appium:udid", "emulator-5554"); //special emulator id
+
+        //required desired capabilities to launch the particular application
+        desiredCapabilities.setCapability("appPackage", "com.android.dialer");
+
+        desiredCapabilities.setCapability("appActivity", "com.android.dialer.main.impl.MainActivity");
+
+        //appium server url
+        URL remoteUrl = new URL(excel.getData(0,0,0));
+
+        driver = new AndroidDriver(remoteUrl, desiredCapabilities);
+
+
+    }
+
+    @Test
+    public void sampleTest() {
```

Comment:
Follow Java naming conventiosn best practices, sampleTest, el1, el2 these are not correct variable naming conventions.